### DISCUSSION TITLE
Thunder Client- Subtract decimal numbers does not gives me values after decimal using subtract filter

### DISCUSSION THREAD

**Discussion by DeltaUser121** [thunderclient/thunder-client-support#1224] [Q&A]
I am using the below in my json query
`{{num1 | subtract(Amount)}}`

where if below are defined in env variables as-
num1=123.43
Amount=2.00

the result I get is 121 but it should be 121.43. How to get the accurate result here?

**Comment by rangav** [comment]
Hi @DeltaUser121

Will verify and get back to you.

**Comment by rangav** [comment]
I have verified it, its a bug,

Please create a bug request here, I will fix it in the next update.
https://github.com/rangav/thunder-client-support/issues

**Reply by testmonger** [reply]
I am using this syntax:
`{{{{accountAvailableBalance-0}} | subtract({{transactionAmount}})}}`

but it is returning a failed assertion with:
Query: `json.results[0].xxx equal to {{75.0500 | subtract(-0.1)}}` => `Actual: 74.9500`

**Reply by rangav** [reply]
{{75.0500 | subtract(-0.1)}} this syntax is worng

you cannot use the number directly it should be variables

`{{var1 | subtract(-0.1)}}`

**Reply by testmonger** [reply]
thank you for the clarification. I need to tweak my assertion some mopre, but I see how it is working now. :)

this:
{{accountAvailableBalance-0 | subtract(transactionAmount)}}
returns this:
Query: json.results[0].xxx equal to 75.14999999999999 => Actual: 74.9500

and this:
{{accountAvailableBalance-0 | add(transactionAmount)}}

returns:
Query: json.results[0].xxx equal to 74.95 => Actual: 74.9500

final syntax here:
{{accountAvailableBalance-0 | add(transactionAmount)}}00

result:
Query: json.results[0].xxx equal to 74.9500 (Pass)
